<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.31">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jonas Schaub">
<meta name="dcterms.date" content="2025-04-23">

<title>NaFM: Pre-training a Foundation Model for Small-Molecule Natural Products (Ding et al. 2025) â€“ Lamalab Tool and Paper Notes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../papers/chen_reasoning.html" rel="next">
<link href="../papers/paper_bench_2025.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-e1a5c8363afafaef2c763b6775fbf3ca.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-6e4858b4921ffe5a9c7d897cee0e5447.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../papers/chemical_reaction_acceleration.html">Papers</a></li><li class="breadcrumb-item"><a href="../papers/NaFM_2025.html"><span class="chapter-title">NaFM: Pre-training a Foundation Model for Small-Molecule Natural Products <span class="citation" data-cites="ding_nafm_2025">(</span></span></a><a href="#ref-ding_nafm_2025" role="doc-biblioref">Ding et al. 2025</a>)</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Lamalab Tool and Paper Notes</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Tool and paper minutes</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Tools</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tools/hydra.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Hydra</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tools/ip_rotator.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">IP Rotator</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tools/polars.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Polars</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tools/thunder_client.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Thunder Client</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tools/tmux.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">tmux</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tools/trimean.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Robust statistics and Trimean</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tools/pandarallel.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Easy fast <code>.apply</code> for pandas</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tools/bfg.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">BFG Repo-Cleaner</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../tools/showyourwork.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">showyourwork</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Papers</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/chemical_reaction_acceleration.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Accelerating the inference of string generation-based chemical reaction models for industrial applications</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/llms_review.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">LLMs in Chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/llm4mat.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Leveraging language representation for materials exploration and discovery</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/chem_yield_prediction_2024.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Uncertainty-Aware Yield Prediction with Multimodal Molecular Features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/dagdelen_data_extraction.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Structured information extraction from scientific text with large language models</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/MolCLR_2024.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Molecular contrastive learning of representations (MolCLR) via graph neural networks</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/PaCh.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">PaCh (Packed Chemicals): Computationally Effective Binary Format for Chemical Structure Encoding</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/ms_prediction_graph_transformers/tandem_ms_prediction_graph_transformers.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Tandem mass spectrum prediction for small molecules using graph transformers</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/open-source-cheminformatics.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Open-Source Software Development in Cheminformatics: A Qualitative Analysis of Rationales</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/universal_chiral_fp_2024.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">MAP4C: One chiral fingerprint to find them all <span class="citation" data-cites="orsi2024one">(</span></span></a><a href="#ref-orsi2024one" role="doc-biblioref">Orsi and Reymond 2024</a>)
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/lwreg_2024.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">lwreg: A Lightweight System for Chemical Registration and Data Storage <span class="citation" data-cites="landrum_lwreg_2024">(</span></span></a><a href="#ref-landrum_lwreg_2024" role="doc-biblioref">Landrum et al. 2024</a>)
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/tango.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">It Takes Two to Tango: Directly Optimizing for Constrained Synthesizability in Generative Molecular Design <span class="citation" data-cites="https://doi.org/10.48550/arxiv.2410.11527">(</span></span></a><a href="#ref-https://doi.org/10.48550/arxiv.2410.11527" role="doc-biblioref">Guo and Schwaller 2024</a>)
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/core_bench.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">CORE-Bench: Fostering the Credibility of Published Research Through a Computational Reproducibility Agent Benchmark</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/bpm_L.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">A Bond-Based Machine Learning Model for Molecular Polarizabilities and A Priori Raman Spectra</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/coconut.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Training Large Language Models to Reason in a Continuous Latent Space</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/emnpd.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">ğŸ§¬ EMNPD Database</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/light-rag.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">ğŸš€ LightRAG</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/nmr-shift-prediction-from-small-data-quantities.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">NMR shift prediction from small data quantities</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/llm_confidence_2025.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">What large language models know and what people think they know</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/paper_bench_2025.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">PaperBench: Evaluating AIâ€™s Ability to Replicate AI Research</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/NaFM_2025.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">NaFM: Pre-training a Foundation Model for Small-Molecule Natural Products <span class="citation" data-cites="ding_nafm_2025">(</span></span></a><a href="#ref-ding_nafm_2025" role="doc-biblioref">Ding et al. 2025</a>)
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/chen_reasoning.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Reasoning Models Donâ€™t Always Say What They Think</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../papers/chemdfm.html" class="sidebar-item-text sidebar-link"><span class="chapter-title"></span></a><a href="https://arxiv.org/pdf/2401.14818">ChemDFM: A Large Language Foundation Model for Chemistry</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#why-did-i-choose-this-paper" id="toc-why-did-i-choose-this-paper" class="nav-link active" data-scroll-target="#why-did-i-choose-this-paper">Why did I choose this paper?</a></li>
  <li><a href="#abstract" id="toc-abstract" class="nav-link" data-scroll-target="#abstract">Abstract</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#model-architecture-and-pre-training" id="toc-model-architecture-and-pre-training" class="nav-link" data-scroll-target="#model-architecture-and-pre-training">Model Architecture and Pre-Training</a>
  <ul class="collapse">
  <li><a href="#general-down-stream-fine-tuning-approach" id="toc-general-down-stream-fine-tuning-approach" class="nav-link" data-scroll-target="#general-down-stream-fine-tuning-approach">General Down-stream Fine-tuning Approach</a></li>
  </ul></li>
  <li><a href="#performance-evaluation-on-downstream-tasks" id="toc-performance-evaluation-on-downstream-tasks" class="nav-link" data-scroll-target="#performance-evaluation-on-downstream-tasks">Performance Evaluation on Downstream Tasks</a>
  <ul class="collapse">
  <li><a href="#np-taxonomy-classification" id="toc-np-taxonomy-classification" class="nav-link" data-scroll-target="#np-taxonomy-classification">NP Taxonomy Classification</a></li>
  <li><a href="#prediction-of-source-organism-kingdom" id="toc-prediction-of-source-organism-kingdom" class="nav-link" data-scroll-target="#prediction-of-source-organism-kingdom">Prediction of Source Organism Kingdom</a></li>
  <li><a href="#prediction-of-np-producing-biosynthetic-gene-cluster-bgc-and-protein-family" id="toc-prediction-of-np-producing-biosynthetic-gene-cluster-bgc-and-protein-family" class="nav-link" data-scroll-target="#prediction-of-np-producing-biosynthetic-gene-cluster-bgc-and-protein-family">Prediction of NP-producing Biosynthetic Gene Cluster (BGC) and protein family</a></li>
  <li><a href="#predicting-bioactivity-for-specific-targets" id="toc-predicting-bioactivity-for-specific-targets" class="nav-link" data-scroll-target="#predicting-bioactivity-for-specific-targets">Predicting bioactivity for specific targets</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion">Conclusion</a></li>
  <li><a href="#personal-comments-questions-to-discuss" id="toc-personal-comments-questions-to-discuss" class="nav-link" data-scroll-target="#personal-comments-questions-to-discuss">Personal comments / questions to discuss</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../papers/chemical_reaction_acceleration.html">Papers</a></li><li class="breadcrumb-item"><a href="../papers/NaFM_2025.html"><span class="chapter-title">NaFM: Pre-training a Foundation Model for Small-Molecule Natural Products <span class="citation" data-cites="ding_nafm_2025">(</span></span></a><a href="#ref-ding_nafm_2025" role="doc-biblioref">Ding et al. 2025</a>)</li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">NaFM: Pre-training a Foundation Model for Small-Molecule Natural Products <span class="citation" data-cites="ding_nafm_2025">(<a href="#ref-ding_nafm_2025" role="doc-biblioref">Ding et al. 2025</a>)</span></span></h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Jonas Schaub </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 23, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="why-did-i-choose-this-paper" class="level2">
<h2 class="anchored" data-anchor-id="why-did-i-choose-this-paper">Why did I choose this paper?</h2>
<ul>
<li>Was curious about what a â€œFoundation Modelâ€ in Natural Product Chemistry would be exactly, what could it be used for?</li>
</ul>
</section>
<section id="abstract" class="level2">
<h2 class="anchored" data-anchor-id="abstract">Abstract</h2>
<ul>
<li>Statement: existing deep learning methods for Natural Products (NP) research primarily rely on supervised learning approaches designed for specific downstream tasks</li>
<li>Statement 2: one-model-for-a-task paradigm often lacks generalizability and leaves significant room for performance improvement</li>
<li>Statement 3: existing models for chemical problems do not work well for NP</li>
<li>This article describes NaFM (â€œNatural product Foundation Modelâ€), a â€œpre-trained foundation modelâ€ for natural products</li>
<li>Pre-training was done using masked graph learning and contrastive learning (see below)</li>
<li>â€œValidationâ€ of the approach was done in 3 experiments using NaFM as a (kind-of) encoder and a Multi-Layer Perceptron (MLP) as a (kind-of) decoder to solve the following tasks:
<ul>
<li>NP taxonomy classification</li>
<li>Predicting biological sources of NP</li>
<li>Predicting bioactivity of NP</li>
</ul></li>
</ul>
</section>
<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<ul>
<li>NP-based drug discovery/design with traditional methods (NP extracts from (cultivated) organisms, bioactivity-guided fractionation, target validation, structure elucidation, etc.) is costly and time-consuming</li>
<li>With todayâ€™s computing resources and databases like COCONUT, LOTUS, and Supernatural II, NP drug discovery can be sped up, perspectively</li>
<li>Prior work:
<ul>
<li><span class="citation" data-cites="xu_composite_2024">(<a href="#ref-xu_composite_2024" role="doc-biblioref">Xu et al. 2024</a>)</span>: predicting kingdom of NP origin species</li>
<li><span class="citation" data-cites="stokes_deep_2020">(<a href="#ref-stokes_deep_2020" role="doc-biblioref">Stokes et al. 2020</a>)</span>: predicting antibiotic activities</li>
<li><span class="citation" data-cites="kim_npclassifier_2021">(<a href="#ref-kim_npclassifier_2021" role="doc-biblioref">Kim et al. 2021</a>)</span>: predicting structural classes and source pathways of NP</li>
<li>Mostly use traditional fingerprints like Morgan/ECFP (Extended Connectivity FingerPrints)</li>
<li>â€œthese methods could achieve satisfying performance but always suffer when applied to out-of-distribution test data or less-related downstream tasks.â€</li>
<li>They exclusively employ supervised learning but this is problematic because we do not have a lot of labelled data, and it is often biased (e.g.&nbsp;plant NP are usually overrepresented in NP databases)</li>
</ul></li>
<li>Pre-training on unlabelled data:
<ul>
<li>Historically, often done with SMILES as data, RNN or Transformers as model architectures, and masked language modelling as a pre-training strategy</li>
<li>Lately, Graph Neural Networks (GNN) as model architectures, hence graphs as input data, and pre-training strategies specifically for molecular graphs seem promising</li>
</ul></li>
<li>But no pre-trained foundation model for NP has been developed so far</li>
<li>Some thoughts on NP structures in general:
<ul>
<li>NP structures in general are more complex and diverse than synthetic compounds, have specific scaffolds (= ring systems) and functional groups; which is why models trained on synthetic compounds usually struggle with NP</li>
<li>But the advantage of NP structures is that they are â€œevolvedâ€, i.e.&nbsp;they often carry â€œpreservedâ€ substructures like scaffolds which depend on / relate to the source pathway or source organism</li>
<li>And these scaffolds are related to the exhibited bioactivity</li>
<li>A NP-specific pre-training strategy should therefore focus on scaffolds</li>
</ul></li>
</ul>
<p><br>
<img src="NaFM_2025_images/Fig1d.png" class="img-fluid quarto-figure quarto-figure-center" width="600" alt="Figure 1 D. The â€œcentral dogmaâ€ of natural products. The biological source, biosynthetic gene clusters, biosynthetic pathways, and bioactivity of natural products are interconnected through the scaffold, which acts as a bridge linking these three key aspects. source: (Ding et al. 2025) CC-BY 4.0"><br>
</p>
<ul>
<li>Contrastive learning, masked learning and activity cliffs
<ul>
<li>Contrastive learning: in the learned embeddings, similar structures should have similar embeddings, and dissimilar structures should have dissimilar embeddings</li>
<li>Masked learning: substructures are â€œmaskedâ€, i.e.&nbsp;bond or atom attributes are omitted and have to be predicted by the model based on the context</li>
<li>Existing approaches use the masking of structures simply as data augmentation and define similarity/dissimilarity for contrastive learning with pairs of the original structures</li>
<li>Activity cliffs: very similar structures can have completely different (bio-)activities</li>
<li>Activity cliffs contradict the principles of contrastive learning</li>
<li>Idea pursued to avoid this here: use the masked structures as â€œsimilarâ€ structures to their original, unmasked structures</li>
<li>Also: mask whole substructures, not just single atoms and bonds, to make pre-training more complex and hence more successful</li>
</ul></li>
</ul>
</section>
<section id="model-architecture-and-pre-training" class="level2">
<h2 class="anchored" data-anchor-id="model-architecture-and-pre-training">Model Architecture and Pre-Training</h2>
<ul>
<li>NaFM: a pre-training framework (â€œpre-trained foundation modelâ€?) specific to NP</li>
<li>Two novel pre-training approaches: scaffold-subgraph reconstruction and scaffold-aware contrastive learning.</li>
</ul>
<p><br>
<img src="NaFM_2025_images/Fig1b.png" class="img-fluid quarto-figure quarto-figure-center" width="600" alt="Figure 1 B. Details of scaffold-subgraph reconstruction. First, a subgraph is randomly selected from the scaffold, consisting of multiple atoms and chemical bonds. In the subgraph, both node and edge attributes are masked, and all nodes within the subgraph are fully connected (i.e., expanded into a fully connected graph), thereby masking the topological information. During the reconstruction process, the model needs to predict both node and edge attributes, while also distinguishing between real edges and those artificially added virtual edges. source: (Ding et al. 2025) CC-BY 4.0"><br>
</p>
<ul>
<li>scaffold-subgraph reconstruction:
<ul>
<li>(not really evident in the figure, but) masking employed only to substructures of the scaffold of an NP!</li>
<li>atom information (elements / atom types, formal charge, chirality) is masked</li>
<li>bond information (multiplicity, direction (stereochemistry)) is masked</li>
<li>topological information is masked, all bonds in the substructure are removed and instead, the atoms become fully connected</li>
</ul></li>
</ul>
<p><br>
<img src="NaFM_2025_images/Fig1c.png" class="img-fluid quarto-figure quarto-figure-center" width="600" alt="Figure 1 C. Details of scaffold-aware contrastive learning. Different colors represent varying weights defined by scaffold similarity. source: (Ding et al. 2025) CC-BY 4.0"><br>
</p>
<ul>
<li>scaffold-aware contrastive learning:
<ul>
<li>For every original input structure, its masked form is set as positive example of a â€œsimilarâ€ structure that should be encoded similarly</li>
<li>the other molecules and their masked forms are set as negative examples, so â€œdissimilarâ€ structures that should be encoded dissimilarly</li>
<li>scaffold similarity is used as a metric for similarity of the complete structures
<ul>
<li>cosine similarity between the MACCS fingerprints of scaffolds is incorporated into the contrastive learning loss function</li>
</ul></li>
<li>This way, scaffold-subgraph reconstruction and scaffold-aware contrastive learning are combined</li>
</ul></li>
</ul>
<p><br>
<img src="NaFM_2025_images/Fig1a.png" class="img-fluid quarto-figure quarto-figure-center" width="600" alt="Figure 1 A. Overview of NaFM pre-training. After the natural product molecule undergoes scaffold-aware masking, both the masked and unmasked information are simultaneously input into a multi-layer Graph Isomorphism Network. The masked information is then processed through a projection head to predict the masked atom attributes, bond attributes, and topological information. Meanwhile, the masked and unmasked information passes through a pooling function and is used for scaffold-aware contrastive learning. source: (Ding et al. 2025) CC-BY 4.0"><br>
</p>
<ul>
<li>Training data: COCONUT NP (0.6 mio molecules) (<span class="citation" data-cites="chandrasekhar_coconut_2024">(<a href="#ref-chandrasekhar_coconut_2024" role="doc-biblioref">Chandrasekhar et al. 2024</a>)</span>)</li>
<li>Model architecture: Graph Neural Network (GNN) / Graph Isomorphism Network (GIN) (6 layers, hidden dimensionality of 512)</li>
<li>Molecule goes in once as original structure and once as scaffold-masked structure</li>
<li>Average pooling operation as the readout function to extract a global molecular representation of 1024 dimensions</li>
<li>Projection head (3 Multi-Layer Perceptrons (MLP)) has to reconstruct/predict the masked properties (atom properties, bond properties, topologies)</li>
<li>a MLP has to embed the original structures and their masked forms (in 256 dimensions) so that original-masked pairs are seen as similar and other structures as dissimilar, based on scaffold similarity (scaffold-aware contrastive learning)</li>
</ul>
<section id="general-down-stream-fine-tuning-approach" class="level3">
<h3 class="anchored" data-anchor-id="general-down-stream-fine-tuning-approach">General Down-stream Fine-tuning Approach</h3>
<ul>
<li>Original contrastive learning head and reconstruction head are replaced with a randomly initialized MLP</li>
<li>For all pre-trained baselines (other models that NaFM is compared to below) the same settings as described above are used</li>
<li>For baselines trained from scratch, their default configurations are used.</li>
</ul>
</section>
</section>
<section id="performance-evaluation-on-downstream-tasks" class="level2">
<h2 class="anchored" data-anchor-id="performance-evaluation-on-downstream-tasks">Performance Evaluation on Downstream Tasks</h2>
<section id="np-taxonomy-classification" class="level3">
<h3 class="anchored" data-anchor-id="np-taxonomy-classification">NP Taxonomy Classification</h3>
<ul>
<li>Data: NPClassifier dataset (<span class="citation" data-cites="kim_npclassifier_2021">(<a href="#ref-kim_npclassifier_2021" role="doc-biblioref">Kim et al. 2021</a>)</span>)
<ul>
<li>Biosynthetic pathway (7 types), super class (70), and class (563) annotations of 77,000 NP structures</li>
</ul></li>
<li>First experiment on class level
<ul>
<li>Multiple pre-trained (general, not NP-specific!) molecular models were used as baselines</li>
<li>Additionally, Extended Connectivity Fingerprints (ECFP) were used as baseline (authors state that they are currently the most popular choice for encoding NP structures)</li>
<li>All baseline models, ECFP, and NaFM were used for training a MLP as described above for NP classification</li>
<li>Results in Table 1</li>
<li>NaFM shows top scores for all dataset sizes, especially for small datasets</li>
</ul></li>
</ul>
<p>Table 1 The AUPRC results for natural product taxonomy classification compared to other baselines.</p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th>Training samples per Class</th>
<th>4</th>
<th>8</th>
<th>16</th>
<th>24</th>
<th>40</th>
<th>64</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>N-Gram</td>
<td>44.72 Â± 1.91</td>
<td>56.61 Â± 0.66</td>
<td>66.36 Â± 1.57</td>
<td>71.11 Â± 0.77</td>
<td>73.34 Â± 1.14</td>
<td>75.77 Â± 0.54</td>
</tr>
<tr class="even">
<td>PretrainGNN</td>
<td>44.83 Â± 0.82</td>
<td>61.85 Â± 0.78</td>
<td>75.76 Â± 0.62</td>
<td>80.50 Â± 0.17</td>
<td>85.31 Â± 0.36</td>
<td>87.82 Â± 0.13</td>
</tr>
<tr class="odd">
<td>D-MPNN</td>
<td>46.63 Â± 0.23</td>
<td>60.88 Â± 0.48</td>
<td>75.73 Â± 0.12</td>
<td>80.96 Â± 0.24</td>
<td>86.64 Â± 0.44</td>
<td>89.23 Â± 0.79</td>
</tr>
<tr class="even">
<td>MolCLR</td>
<td>45.76 Â± 1.98</td>
<td>65.80 Â± 1.51</td>
<td>78.14 Â± 1.08</td>
<td>83.20 Â± 0.74</td>
<td>85.56 Â± 0.11</td>
<td>88.22 Â± 0.34</td>
</tr>
<tr class="odd">
<td>Mole-BERT</td>
<td>66.32 Â± 1.23</td>
<td>73.39 Â± 0.66</td>
<td>78.25 Â± 0.45</td>
<td>80.83 Â± 0.51</td>
<td>83.57 Â± 0.59</td>
<td>85.69 Â± 0.70</td>
</tr>
<tr class="even">
<td>ECFP</td>
<td>69.17 Â± 0.19</td>
<td>78.21 Â± 0.79</td>
<td>83.82 Â± 0.37</td>
<td>86.28 Â± 0.49</td>
<td>88.52 Â± 0.48</td>
<td>89.75 Â± 0.45</td>
</tr>
<tr class="odd">
<td>GraphMVP</td>
<td>64.50 Â± 0.74</td>
<td>78.41 Â± 0.09</td>
<td>85.71 Â± 0.43</td>
<td>87.88 Â± 0.26</td>
<td>89.72 Â± 0.45</td>
<td>91.07 Â± 0.43</td>
</tr>
<tr class="even">
<td><strong>NaFM</strong></td>
<td>70.10 Â± 0.92</td>
<td>79.89 Â± 0.07</td>
<td>87.37 Â± 1.51</td>
<td>89.15 Â± 0.22</td>
<td>90.77 Â± 0.26</td>
<td>91.75 Â± 0.47</td>
</tr>
</tbody>
</table>
<ul>
<li>Second experiment comparison to NPClassifier
<ul>
<li>NPClassifier was taken as-is, NaFM was fine-tuned as described above</li>
<li>comparison on pathway and super-class levels</li>
<li>Results in Figure 2 (PR AUC: Area Under the Curve where x is recall and y is precision)</li>
<li>The fine-tuned NaFM outperforms NPClassifier</li>
<li>Especially for amino acids and peptides (pathway) and Carotenoids (C45), Tetramate alkaloids, Peptide alkaloids, and Anthranilic acid alkaloids (super-classes)</li>
<li>The gap in the recognition of Î³-lactam-Î²-lactones can be attributed to the limited data for this category, with only 8 instances out of 78,000 data points</li>
</ul></li>
</ul>
<p><br>
<img src="NaFM_2025_images/Fig2.png" class="img-fluid quarto-figure quarto-figure-center" width="600" alt="Figure 2. The bar plots illustrate the modelsâ€™ performance comparison of NaFM and NPClassifier on various superclass categories (a) and biosynthetic pathways (b) including Carbohydrates, Amino Acids and Peptides, Alkaloids, Terpenoids, Shikimates and Phenylpropanoids, Polyketides, and Fatty Acids in terms of AUPRC. The gray segments indicate the shared performance (minimum AUPRC) and the light blue/orange segments highlight the additional AURPC achieved by NaFM and NPClassifier beyond each other, respectively. Higher values indicate better model performance. source: (Ding et al. 2025) CC-BY 4.0"><br>
</p>
</section>
<section id="prediction-of-source-organism-kingdom" class="level3">
<h3 class="anchored" data-anchor-id="prediction-of-source-organism-kingdom">Prediction of Source Organism Kingdom</h3>
<ul>
<li>Based on LOTUS (<span class="citation" data-cites="rutz_lotus_2022">(<a href="#ref-rutz_lotus_2022" role="doc-biblioref">Rutz et al. 2022</a>)</span>)
<ul>
<li>130,000 data pairs of NP structure and source organism (and literature reference)</li>
</ul></li>
<li>NaFM model was fine-tuned to predict kingdom of NP-producing species (i.e.&nbsp;animals, bacteria, <a href="https://ucmp.berkeley.edu/chromista/chromista.html">chromista</a>, fungi, and plants)</li>
<li>Figure 3 shows the resulting embeddings plotted using TMAP <span class="citation" data-cites="probst_visualization_2020">(<a href="#ref-probst_visualization_2020" role="doc-biblioref">Probst and Reymond 2020</a>)</span></li>
<li>NaFM appears to be able to separate the NP of the different kingdoms within the representation space</li>
<li>Kingdom representative structures were obtained by performing a cluster analysis and giving the representative of the biggest cluster, respectively</li>
<li>Without the pre-training weights (â€œfrom scratchâ€), the model is not able to separate the classes (larger figure available in SI)</li>
</ul>
<p><br>
<img src="NaFM_2025_images/Fig3.png" class="img-fluid quarto-figure quarto-figure-center" width="600" alt="Figure 3. The obtained representations are projected into two dimensions using TMAP and visualized with Faerun. For each biological source, a representative example is highlighted in a designated box, showcasing a characteristic scaffold that typifies compounds derived from that particular source. source: (Ding et al. 2025) CC-BY 4.0"><br>
</p>
</section>
<section id="prediction-of-np-producing-biosynthetic-gene-cluster-bgc-and-protein-family" class="level3">
<h3 class="anchored" data-anchor-id="prediction-of-np-producing-biosynthetic-gene-cluster-bgc-and-protein-family">Prediction of NP-producing Biosynthetic Gene Cluster (BGC) and protein family</h3>
<ul>
<li>dataset a combination of MIBiG (Minimum Information about a Biosynthetic Gene cluster database, <span class="citation" data-cites="terlouw_mibig_2023">(<a href="#ref-terlouw_mibig_2023" role="doc-biblioref">Terlouw et al. 2023</a>)</span>) and Pfam (Protein families database, <span class="citation" data-cites="mistry_pfam_2021">(<a href="#ref-mistry_pfam_2021" role="doc-biblioref">Mistry et al. 2021</a>)</span>): 2,000 bacterial and fungal BGCs and their corresponding metabolite structures, plus protein families encoded by the respective BGC</li>
<li>Again, NaFM, ECFP, and baseline molecular pre-trained models were fine-tuned on the task of predicting the producing protein family based on an NP structure</li>
<li>Results in Figure 4 for the top 128 protein families, NaFM again outperforms all others (AUROC: Area Under ROC curve)
<ul>
<li>No idea why they call it NaRel instead of NaFM hereâ€¦</li>
<li>e.g.&nbsp;ID 65 and 115 are protein families are known to be highly conserved</li>
</ul></li>
</ul>
<p><br>
<img src="NaFM_2025_images/Fig4.png" class="img-fluid quarto-figure quarto-figure-center" width="600" alt="Figure 4. The figure visualizes the AUROC gap between NaFM and various molecular representations. The results show the AUROC for the 128 most frequent protein families found in bacteria and fungi. The color in each block represents the magnitude of the AUROC gap, ranging from negative to positive, with colors transitioning from red to blue. Deeper colors indicate a greater absolute difference. source: (Ding et al. 2025) CC-BY 4.0"><br>
</p>
</section>
<section id="predicting-bioactivity-for-specific-targets" class="level3">
<h3 class="anchored" data-anchor-id="predicting-bioactivity-for-specific-targets">Predicting bioactivity for specific targets</h3>
<ul>
<li>data: human protein targets and HIV type- 1 reverse transcriptase activity data from the NPASS natural product activity database</li>
<li>As before, NaFM and the baseline models were fine-tuned on this data</li>
<li>Results in Table 2</li>
<li>NaFM surpasses all baselines, except for N-Gram on HIV type-1 reverse transcriptase</li>
</ul>
<p>Table 2 The RMSE results for natural product bioactivity regression compared to other baselines.</p>
<table class="caption-top table">
<colgroup>
<col style="width: 8%">
<col style="width: 10%">
<col style="width: 11%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 11%">
</colgroup>
<thead>
<tr class="header">
<th>Target Name</th>
<th>Number of samples</th>
<th>N-Gram</th>
<th>PretrainGNN</th>
<th>D-MPNN</th>
<th>MolCLR</th>
<th>Mol-BERT</th>
<th>ECFP</th>
<th>GraphMVP</th>
<th>NaFM</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PTP1B</td>
<td>612</td>
<td>0.8493 Â± 0.2623</td>
<td>0.8652 Â± 0.2110</td>
<td>0.8577 Â± 0.2165</td>
<td>0.9912 Â± 0.2467</td>
<td>0.8624 Â± 0.0732</td>
<td>1.1409 Â± 0.2281</td>
<td>1.8865 Â± 0.1303</td>
<td><strong>0.8243 Â± 0.1960</strong></td>
</tr>
<tr class="even">
<td>AChE</td>
<td>341</td>
<td>1.3774 Â± 0.3995</td>
<td>1.5462 Â± 0.2265</td>
<td>1.2881 Â± 0.1992</td>
<td>1.1507 Â± 0.1683</td>
<td>1.2938 Â± 0.3615</td>
<td>1.3721 Â± 0.0703</td>
<td>2.8140 Â± 0.3649</td>
<td><strong>1.1227 Â± 0.1604</strong></td>
</tr>
<tr class="odd">
<td>COX-2</td>
<td>190</td>
<td>1.0022 Â± 0.3192</td>
<td>1.7231 Â± 0.3754</td>
<td>1.0272 Â± 0.2468</td>
<td>0.9399 Â± 0.1705</td>
<td>1.0529 Â± 0.1536</td>
<td>1.2718 Â± 0.1531</td>
<td>3.0341 Â± 0.6418</td>
<td><strong>0.9239 Â± 0.1721</strong></td>
</tr>
<tr class="even">
<td>HIV type-1 RT</td>
<td>186</td>
<td><strong>1.0606 Â± 0.1613</strong></td>
<td>1.5210 Â± 0.2606</td>
<td>1.1826 Â± 0.1683</td>
<td>1.2649 Â± 0.1353</td>
<td>1.4348 Â± 0.1398</td>
<td>1.5446 Â± 0.1133</td>
<td>2.7829 Â± 0.1195</td>
<td>1.0802 Â± 0.1506</td>
</tr>
<tr class="odd">
<td>Tyrosinase</td>
<td>186</td>
<td>0.7822 Â± 0.1408</td>
<td>1.6957 Â± 0.2915</td>
<td>0.8206 Â± 0.1891</td>
<td>0.9165 Â± 0.2820</td>
<td>1.0440 Â± 0.2569</td>
<td>1.2553 Â± 0.0985</td>
<td>3.3965 Â± 0.3437</td>
<td><strong>0.6927 Â± 0.2828</strong></td>
</tr>
<tr class="even">
<td>CYP3A4</td>
<td>178</td>
<td>0.7846 Â± 0.2037</td>
<td>1.4157 Â± 0.2923</td>
<td>0.6946 Â± 0.1415</td>
<td>0.7072 Â± 0.1362</td>
<td>0.9377 Â± 0.1873</td>
<td>1.0439 Â± 0.1244</td>
<td>3.0817 Â± 0.2720</td>
<td><strong>0.6922 Â± 0.1326</strong></td>
</tr>
<tr class="odd">
<td>MRP4</td>
<td>177</td>
<td>0.2747 Â± 0.0846</td>
<td>1.6869 Â± 0.2476</td>
<td>0.2399 Â± 0.0603</td>
<td>0.2321 Â± 0.0847</td>
<td>0.5190 Â± 0.0645</td>
<td>1.4682 Â± 0.2422</td>
<td>3.5723 Â± 0.4337</td>
<td><strong>0.2265 Â± 0.0921</strong></td>
</tr>
<tr class="even">
<td>COX-1</td>
<td>140</td>
<td>0.9529 Â± 0.1643</td>
<td>2.6831 Â± 0.2056</td>
<td>0.9405 Â± 0.1414</td>
<td>0.9574 Â± 0.1655</td>
<td>1.4542 Â± 0.3456</td>
<td>1.1965 Â± 0.1850</td>
<td>3.0083 Â± 0.2802</td>
<td><strong>0.9286 Â± 0.1342</strong></td>
</tr>
</tbody>
</table>
<ul>
<li>Second experiment in this area:
<ul>
<li>acetylcholinesterase(AChE) as the target</li>
<li>data: AChE IC50 data from the NPASS database</li>
<li>NaFM model is fine-tuned on this data</li>
<li>Fine-tuned model is used to predict AChE activity for all COCONUT NP</li>
<li>The 600 molecules with the best predicted activity are tested in virtual screening (docking)</li>
<li>as control, 600 randomly chosen NP are also tested</li>
<li>Figure 5 a: docking scores for the two groups
<ul>
<li>the two distributions are significantly different, the NP with predicted activity are significantly better than the randomly chosen ones</li>
<li>Model-selected NP scores have a higher mean and lower variance</li>
</ul></li>
</ul></li>
</ul>
<p><br>
<img src="NaFM_2025_images/Fig5a.png" class="img-fluid quarto-figure quarto-figure-center" width="600" alt="Figure 5a. Docking scores of model ranked top 0.1% (approximately 600 molecules) of acetylcholinesterase inhibitors, along with 600 randomly selected molecules. The histogram and distribution of docking scores with acetylcholinesterase are shown in the figure. Orange represents the model-selected group, while blue represents the random group. The x-axis represents the docking score, and the y-axis represents frequency. A lower docking score (more left-shifted distribution) indicates higher inhibitory activity. Docking was performed using SchrÃ¶dinger Maestro, with detailed settings provided in the Supplementary section 1.3. source: (Ding et al. 2025) CC-BY 4.0"><br>
</p>
<ul>
<li>Figure 5c shows some COCONUT NP with high model and docking scores, alongside the most structurally similar NP from the fine-tuning set (NPASS)
<ul>
<li>demonstrated ability of the model to generalise or find novel actives</li>
</ul></li>
</ul>
<p><br>
<img src="NaFM_2025_images/Fig5c.png" class="img-fluid quarto-figure quarto-figure-center" width="600" alt="Figure 5c. The first and second columns display several molecules with high model scores and good docking activity, along with their most similar positive counterparts encountered during the model fine-tuning process. The third column shows the maximum common substructure between these two molecules. source: (Ding et al. 2025) CC-BY 4.0"><br>
</p>
</section>
</section>
<section id="conclusion" class="level2">
<h2 class="anchored" data-anchor-id="conclusion">Conclusion</h2>
<ul>
<li>proposed NaFM, a foundation model for small molecule natural products</li>
<li>Generated using novel pre-training strategies</li>
<li>NaFM has been evaluated across a range of tasks, including upstream genome mining, biological source identification, structural classification, metabolic pathway prediction, and activity prediction and screening.</li>
<li>On most of the benchmarks, NaFM outperforms both traditional molecular fingerprints and deep-learning based pre-training frameworks</li>
<li>It is expected that NaFM can accelerate the NP drug discovery process in multiple stages</li>
</ul>
</section>
<section id="personal-comments-questions-to-discuss" class="level2">
<h2 class="anchored" data-anchor-id="personal-comments-questions-to-discuss">Personal comments / questions to discuss</h2>
<ul>
<li>There is no code available, apparently; sad</li>
<li>Larger text sections, e.g.&nbsp;the abstract, are quite hard to understand, i.e.&nbsp;there are many â€œniceâ€ words but no details</li>
<li>Is the presented model really a â€œfoundation modelâ€ and not rather an â€œencoderâ€?</li>
</ul>
</section>
<section id="references" class="level2">
<h2 class="anchored" data-anchor-id="references">References</h2>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-chandrasekhar_coconut_2024" class="csl-entry" role="listitem">
Chandrasekhar, Venkata, Kohulan Rajan, Sri Ram Sagar Kanakam, Nisha Sharma, Viktor WeiÃŸenborn, Jonas Schaub, and Christoph Steinbeck. 2024. <span>â€œ<span>COCONUT</span> 2.0: A Comprehensive Overhaul and Curation of the Collection of Open Natural Products Database.â€</span> <em>Nucleic Acids Research</em>, November, gkae1063. <a href="https://doi.org/10.1093/nar/gkae1063">https://doi.org/10.1093/nar/gkae1063</a>.
</div>
<div id="ref-ding_nafm_2025" class="csl-entry" role="listitem">
Ding, Yuheng, Yusong Wang, Bo Qiang, Jie Yu, Qi Li, Yiran Zhou, and Zhenmin Liu. 2025. <span>â€œ<span>NaFM</span>: <span>Pre</span>-Training a <span>Foundation</span> <span>Model</span> for <span>Small</span>-<span>Molecule</span> <span>Natural</span> <span>Products</span>.â€</span> <a href="https://doi.org/10.48550/ARXIV.2503.17656">https://doi.org/10.48550/ARXIV.2503.17656</a>.
</div>
<div id="ref-https://doi.org/10.48550/arxiv.2410.11527" class="csl-entry" role="listitem">
Guo, Jeff, and Philippe Schwaller. 2024. <span>â€œIt Takes Two to Tango: Directly Optimizing for Constrained Synthesizability in Generative Molecular Design.â€</span> arXiv. <a href="https://doi.org/10.48550/ARXIV.2410.11527">https://doi.org/10.48550/ARXIV.2410.11527</a>.
</div>
<div id="ref-kim_npclassifier_2021" class="csl-entry" role="listitem">
Kim, Hyun Woo, Mingxun Wang, Christopher A. Leber, Louis-FÃ©lix Nothias, Raphael Reher, Kyo Bin Kang, Justin J. J. Van Der Hooft, Pieter C. Dorrestein, William H. Gerwick, and Garrison W. Cottrell. 2021. <span>â€œ<span>NPClassifier</span>: <span>A</span> <span>Deep</span> <span>Neural</span> <span>Network</span>-<span>Based</span> <span>Structural</span> <span>Classification</span> <span>Tool</span> for <span>Natural</span> <span>Products</span>.â€</span> <em>Journal of Natural Products</em> 84 (11): 2795â€“2807. <a href="https://doi.org/10.1021/acs.jnatprod.1c00399">https://doi.org/10.1021/acs.jnatprod.1c00399</a>.
</div>
<div id="ref-landrum_lwreg_2024" class="csl-entry" role="listitem">
Landrum, Gregory A., Jessica Braun, Paul Katzberger, Marc T. Lehner, and Sereina Riniker. 2024. <span>â€œLwreg: <span>A</span> <span>Lightweight</span> <span>System</span> for <span>Chemical</span> <span>Registration</span> and <span>Data</span> <span>Storage</span>.â€</span> <em>Journal of Chemical Information and Modeling</em> 64 (16): 6247â€“52. <a href="https://doi.org/10.1021/acs.jcim.4c01133">https://doi.org/10.1021/acs.jcim.4c01133</a>.
</div>
<div id="ref-mistry_pfam_2021" class="csl-entry" role="listitem">
Mistry, Jaina, Sara Chuguransky, Lowri Williams, Matloob Qureshi, Gustavo A Salazar, Erik L L Sonnhammer, Silvio C E Tosatto, et al. 2021. <span>â€œPfam: <span>The</span> Protein Families Database in 2021.â€</span> <em>Nucleic Acids Research</em> 49 (D1): D412â€“19. <a href="https://doi.org/10.1093/nar/gkaa913">https://doi.org/10.1093/nar/gkaa913</a>.
</div>
<div id="ref-orsi2024one" class="csl-entry" role="listitem">
Orsi, Markus, and Jean-Louis Reymond. 2024. <span>â€œOne Chiral Fingerprint to Find Them All.â€</span> <em>Journal of Cheminformatics</em> 16 (1): 53.
</div>
<div id="ref-probst_visualization_2020" class="csl-entry" role="listitem">
Probst, Daniel, and Jean-Louis Reymond. 2020. <span>â€œVisualization of Very Large High-Dimensional Data Sets as Minimum Spanning Trees.â€</span> <em>Journal of Cheminformatics</em> 12 (1): 12. <a href="https://doi.org/10.1186/s13321-020-0416-x">https://doi.org/10.1186/s13321-020-0416-x</a>.
</div>
<div id="ref-rutz_lotus_2022" class="csl-entry" role="listitem">
Rutz, Adriano, Maria Sorokina, Jakub Galgonek, Daniel Mietchen, Egon Willighagen, Arnaud Gaudry, James G Graham, et al. 2022. <span>â€œThe <span>LOTUS</span> Initiative for Open Knowledge Management in Natural Products Research.â€</span> <em>eLife</em> 11 (May): e70780. <a href="https://doi.org/10.7554/eLife.70780">https://doi.org/10.7554/eLife.70780</a>.
</div>
<div id="ref-stokes_deep_2020" class="csl-entry" role="listitem">
Stokes, Jonathan M., Kevin Yang, Kyle Swanson, Wengong Jin, Andres Cubillos-Ruiz, Nina M. Donghia, Craig R. MacNair, et al. 2020. <span>â€œA <span>Deep</span> <span>Learning</span> <span>Approach</span> to <span>Antibiotic</span> <span>Discovery</span>.â€</span> <em>Cell</em> 180 (4): 688â€“702.e13. <a href="https://doi.org/10.1016/j.cell.2020.01.021">https://doi.org/10.1016/j.cell.2020.01.021</a>.
</div>
<div id="ref-terlouw_mibig_2023" class="csl-entry" role="listitem">
Terlouw, Barbara R, Kai Blin, Jorge C Navarro-MuÃ±oz, Nicole E Avalon, Marc G Chevrette, Susan Egbert, Sanghoon Lee, et al. 2023. <span>â€œ<span>MIBiG</span> 3.0: A Community-Driven Effort to Annotate Experimentally Validated Biosynthetic Gene Clusters.â€</span> <em>Nucleic Acids Research</em> 51 (D1): D603â€“10. <a href="https://doi.org/10.1093/nar/gkac1049">https://doi.org/10.1093/nar/gkac1049</a>.
</div>
<div id="ref-xu_composite_2024" class="csl-entry" role="listitem">
Xu, Qisong, Alan K. X. Tan, Liangfeng Guo, Yee Hwee Lim, Dillon W. P. Tay, and Shi Jun Ang. 2024. <span>â€œComposite Machine Learning Strategy for Natural Products Taxonomical Classification and Structural Insights.â€</span> <em>Digital Discovery</em> 3 (11): 2192â€“2200. <a href="https://doi.org/10.1039/D4DD00155A">https://doi.org/10.1039/D4DD00155A</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "î§‹";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../papers/paper_bench_2025.html" class="pagination-link" aria-label="PaperBench: Evaluating AIâ€™s Ability to Replicate AI Research">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">PaperBench: Evaluating AIâ€™s Ability to Replicate AI Research</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../papers/chen_reasoning.html" class="pagination-link" aria-label="Reasoning Models Donâ€™t Always Say What They Think">
        <span class="nav-page-text"><span class="chapter-title">Reasoning Models Donâ€™t Always Say What They Think</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>